test_util/recordio.py

```diff
@@ -0,0 +1,155 @@
+        self.serialize = serialize
+
+    def encode(self, message):
+        """Encode a message into 'RecordIO' format.
+
+        :param message: a message to serialize and then wrap in
+                        a 'RecordIO' frame.
+        :type message: object
+        :returns: a serialized message wrapped in a 'RecordIO' frame
+        :rtype: bytes
+        """
+
+        s = self.serialize(message)
+
+        if not isinstance(s, bytes):
+            raise DCOSException("Calling 'serialize(message)' must"
+                                " return a 'bytes' object")
+
+        return bytes(str(len(s)) + "\n", "UTF-8") + s
+
+
+class Decoder(object):
+    """Decode a 'RecordIO' message back to an arbitrary message type.
+
+       This class encapsulates the process of decoding a message
+       previously encoded with 'RecordIO' back to an arbitrary
+       message type. Its constructor takes a deserialization
+       function of the form 'deserialize(data)'. This
+       deserialization function is responsible for knowing how to
+       take a fully constructed 'RecordIO' message containing a
+       'UTF-8' encoded byte array and deserialize it back into the
+       original message type.
+
+       The 'decode(data)' message takes a 'UTF-8' encoded byte array
+       as input and buffers it across subsequent calls to
+       construct a set of fully constructed 'RecordIO' messages that
+       are decoded and returned in a list.
+
+       :param deserialize: a function to deserialize from 'RecordIO'
+                           messages built up by subsequent calls
+                           to 'decode(data)'
+       :type deserialize: function
+    """
+
+    HEADER = 0
+    RECORD = 1
+    FAILED = 2
+
+    def __init__(self, deserialize):
+        self.deserialize = deserialize
+        self.state = self.HEADER
+        self.buffer = bytes("", "UTF-8")
+        self.length = 0
+
+    def decode(self, data):
+        """Decode a 'RecordIO' formatted message to its original type.
+
+        :param data: an array of 'UTF-8' encoded bytes that make up a
+                      partial 'RecordIO' message. Subsequent calls to this
+                      function maintain state to build up a full 'RecordIO'
+                      message and decode it
+        :type data: bytes
+        :returns: a list of deserialized messages
+        :rtype: list
+        """
+
+        if not isinstance(data, bytes):
+            raise DCOSException("Parameter 'data' must of of type 'bytes'")
+
+        if self.state == self.FAILED:
+            raise DCOSException("Decoder is in a FAILED state")
+
+        records = []
+
+        for c in data:
+            if self.state == self.HEADER:
+                if c != ord('\n'):
+                    self.buffer += bytes([c])
+                    continue
+
+                try:
+                    self.length = int(self.buffer.decode("UTF-8"))
+                except Exception as exception:
```

Comment:
As this commit is written now, this try/catch/raise DCOSException is never really utilized so when this does get hit, everything will come to a halt anyway, thus, just leave it out.

However, if you envision this being used elsewhere, I could get the idea of a failed state/custom exception. In that case, just make in this module:
```
class RecordIoDecoderException(Exception):
    pass
```
or call it RecordIoCodecError if you want to use one error that also covers the encoder. In either case, those names will be more usable than "DCOSExpcetion" when this has nothing to do with DC/OS on the surface